test/conf_include_test.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP "/\\"
+# define chdir _chdir
+#else
+# include <unistd.h>
```

Comment:
OK, do you have some suggestion what to do? I do not have access to 'all non-Windows platforms' to find out what to include for chdir.

Reply:
I'd suggest guarding it with OPENSSL_SYS_UNIX. If we're not on Unix, and we're not on Windows then don't do that bit of the test.

Reply:
The test does not work at all without chdir.

Reply:
Right - of course. Then skip the test entirely in that case.

Reply:
Is there a way how to not compile the file at all in such case? Or should I put the whole conf_include_test.c into #ifdef ?

Reply:
Not sure (I'll defer to @levitte on that) - but I think probably not. In which case you can just put the whole test inside #ifdef (although, you'll probably have to provide some minimal `setup_tests` implementation which just does nothing.

Reply:
I could not find out how to check for OPENSSL_SYS_UNIX in the perl test recipe for skipping the test. Can you give a hint please?

Reply:
I'm not sure if you can (@levitte?). If you have a minimal `setup_tests` that doesn't add any tests then you don't need to.

Reply:
Ah, so planning and running 3 tests in the recipe won't break the tests if setup_test does not add any test?

Reply:
One very simple way that works everywhere is this:

``` C
# include <openssl/opensslconf.h>
# include OPENSSL_UNISTD
```

Reply:
Everywhere *non-Windows*, that is.  So the current guards are still relevant

Reply:
One can also put it this way. Modulo Windows, is there POSIX[!] platform without unistd.h? Because if there isn't, then NO_POSIX_IO around whole code would do. And I would like to argue that this would be more appropriate. To keep unistd.h inside NO_POSIX_IO that is. There is only one platform in Configurations/*.conf that defines unistd config parameter, nextstep. None of us has ever seen it, and I find it hard to believe that anybody has ever attempted to compile even 1.0.2 on it. It would be more honest to stop pretending that it's supported. Yes, even now. And more or less consciously "break" it by keeping unistd.h. ["break" is in quotes because there is no reason to believe that it's not broken already.] If anybody complains, we would be much better off fixing it in some other way. [On related note I did suggest to degrade nextstep to deprecated].

Reply:
I'll do it as @dot-asm suggested.